jekyll/_cci2/orbs-best-practices.md

```diff
@@ -1,112 +1,226 @@
+      - run: example login $<<parameters.api-token>>
+      - run: example deploy my-app
 ```
 
-### Parameters
+#### Check For Root
+{:.no_toc}
 
-- When possible, use defaults for parameters unless a user input is essential. 
-- Utilize the [“env_var_name” parameter type]({{ site.baseurl }}/2.0/reusing-config/#environment-variable-name) to secure API keys, webhook urls or other sensitive data. 
-- [Injecting steps as a parameter]({{ site.baseurl }}/2.0/reusing-config/#steps) is a useful way to run user defined steps within a job between your orb-defined steps.Good for if you need to perform an action both before and after user-defined tasks - for instance, you could run user-provided steps between your caching logic inside the command.
+Before adding "sudo" to your commands, check to see if the user is already the root user. This can be done dynamically with environment variables.
+
+```bash
+if [[ $EUID == 0 ]]; then export SUDO=""; else # Check if we're root
+  export SUDO="sudo";
+fi
 
-**Installing binaries and tools**
-  - Set an `install-path` parameter, ideally with a default value of `/usr/local/bin`, and ensure to install the binary to this parameterized location. This may often avoid the issue of needing `root` privileges in environments where the user may not have root.
-  - If `root`is required for your use case, it is recommended to add pre-flight checks to determine if the user has root permissions and gracefully fail with a descriptive error message alerting the user they need proper permissions.
-  - Add the binary to the user's path via `$BASH_ENV` so the user may call the binary from a separate [run]({{ site.baseurl }}/2.0/configuration-reference/#run) statement. This is required when installing to a non-default path.
-  example:
+$SUDO do_command
 ```
-echo `export PATH="$PATH:<<parameters.install-path>>"` >> $BASH_ENV
+
+### Jobs
+
+#### Consider "pass-through" Parameters
+{:.no_toc}
+
+Inside your job, if you are utilizing any commands or executors, you must include a copy of each parameter from each of those components into your job. You can then "pass-through" the parameters given to the job, to each referenced component.
+
+For example, here is a partial snippet of the [Node orb's "test" job](https://circleci.com/orbs/registry/orb/circleci/node#jobs-test):
+
+{:.tab.nodeParam.Test_Job}
+```yaml
+description: |
+  Simple drop-in job to test your NodeJS application automatically.
+parameters:
+  version:
+    default: 13.11.0
+    description: >
+      A full version tag must be specified. Example: "13.11.0" For a full list
+      of releases, see the following: https://nodejs.org/en/download/releases
+    type: string
+executor:
+  name: default
+  tag: << parameters.version >>
 ```
 
+{:.tab.nodeParam.Default_Executor}
+```yaml
+description: >
+  Select the version of NodeJS to use. Uses CircleCI's highly cached convenience
+  images built for CI.
+
+  Any available tag from this list can be used:
```

Comment:
link to developer hub maybe?

Reply:
This is directly from the orb executor source itself, which I agree we should probably now change. Let's manually change it here. Really change it in the orb later.